wcfsetup/install/files/ts/WoltLabSuite/Core/Form/Builder/Field/ItemList.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Data handler for an item list form builder field in an Ajax form.
+ *
+ * @author  Matthias Schmidt
+ * @copyright 2001-2020 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Form/Builder/Field/ItemList
+ * @since 5.2
+ */
+
+import Field from "./Field";
+import * as UiItemListStatic from "../../../Ui/ItemList/Static";
+import { FormBuilderData } from "../Data";
+import * as Core from "../../../Core";
+
+class ItemList extends Field {
+  protected _getData(): FormBuilderData {
+    const values: string[] = [];
+    UiItemListStatic.getValues(this._fieldId).forEach((item) => {
+      if (item.objectId) {
+        values[item.objectId] = item.value;
+      } else {
+        values.push(item.value);
+      }
+    });
```

Comment:
I know that this is identical to the old code, but `values[item.objectId] = item.value;` will create a sparse array when the `objectId` is non-consecutive. Should `values` be an object instead?

Reply:
The idea here is to always have an array, even if `!item.objectId`. Additionally, this is existing code so changing it could cause compatibility issues.